### PR TITLE
Prepare 0.20.0 release

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

To prepare for the upcoming 0.20.0 release this commit moves all the
release notes into a 0.20 subdirectory to keep them separate from future
development notes post-release. It also adds a release note with the
prelude section providing the high level overview of the release. When
this merges it should be the commit used to tag the 0.20.0 release.

### Details and comments

- [x] Wait for the remaining open 0.20 PRs to merge or be deferred
- [x] Finish editing and fixing release notes

### CODE REVIEW

File: tox.ini
Comment (mtreinish) [inline tox.ini:68]: I will undo this before we merge, I just need this to build locally, I'm hitting an error when trying to build with 3.10 that I don't feel like debugging right now.

File: releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml
Comment (eliarbel) [inline releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml:4]: I think we should remove this - given that we'll roll-back the usage of `Operation` class for 0.20

Comment (mtreinish) [reply releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml:4]: Ok sure, I'll drop the release note.

Comment (eliarbel) [reply releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml:4]: I removed it here actually: #7840

**Comment by mtreinish** [comment]
I've finished a first pass through the release notes. I still need to review it all again both for typos I likely introduced and any notes I might have missed (along with waiting for the rest of the pending release PRs to merge).

**Comment by jakelishman** [comment]
Ok, I'll finish going through them now - I have a few changes from the ones I looked at last night, but nothing major (obviously).

File: releasenotes/notes/0.20/prepare-0.20-79918ed0fc5b496e.yaml
Comment (ajavadia) [inline releasenotes/notes/0.20/prepare-0.20-79918ed0fc5b496e.yaml:22]:
```suggestion
  * The introduction of the :mod:`qiskit.primitives` module. These APIs are
    provide different abstraction levels for computing outputs of interest from
    :class:`~.QuantumCircuit` and using backends. For
    example, the :class:`~qiskit.primitives.BaseEstimator` defines an abstract
    interface for estimating an expectation value of an observable.
    This can then be used to construct higher level algorithms and applications
    that are built using the estimation of expectation values without having
    to worry about the implementation of computing the expectation value.
    This decoupling allows the implementation to improve in speed and quality
    while adhering to the defined abstract interface.
    Likewise, the :class:`~qiskit.primitives.BaseSampler` computes
    quasi-probability distributions from circuit measurements. Other primitives will
    be introduced in the future.
```

**Review by jakelishman — COMMENTED** [review]
I left my suggested changes in mtreinish/qiskit-core#17

File: releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml
Comment (mtreinish) [reply releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml]: Oh, it should have been dropped in https://github.com/Qiskit/qiskit-terra/pull/7828/commits/cef94f34b11161f8893a7a4d4beb018199f65318

Comment (jakelishman) [reply releasenotes/notes/0.20/Operation-abstract-base-class-99f46180c7ddaa6b.yaml]: It was dropped - I started this review before that commit, and the message must have got lost in the aether.

File: tox.ini
Comment (mtreinish) [reply tox.ini]: Done in: https://github.com/Qiskit/qiskit-terra/pull/7828/commits/0ba05920e7904b272032e39745feaa3b67d2cff4

**Review by jakelishman — APPROVED** [review]
Looks good to me. After the merging of the PR I'd made, I was happy with all the release notes already.

File: releasenotes/notes/0.20/add-xxminusyy-gate-63e6530c23500de9.yaml
Comment (Cryoris) [reply releasenotes/notes/0.20/add-xxminusyy-gate-63e6530c23500de9.yaml:4]: Ah I just saw Kevin beat me to this correction 😄

**Comment by mtreinish** [comment]
Thanks for the suggestions, given the CI time required to apply them I'm going to save these changes fix them manually in the hosted version when we push the release and open up a follow up PR after the release is tagged and backport them for 0.20.1

**Pull request merged**
